xla/service/gpu/nccl_api.h

```diff
@@ -162,7 +163,7 @@ class NcclApi {
   // https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclcommsplit
   virtual absl::StatusOr<std::vector<OwnedNcclComm>> CommSplit(
       absl::Span<const NcclCommHandle> comms, int32_t color,
-      absl::Span<const int32_t> keys) = 0;
+      absl::Span<const int32_t> keys, const Config& config) = 0;
```

Comment:
make it optional? because I think the default is actually to reuse config from parent (because otherwise resources are not shared? are they?)

Reply:
Since in xla we always set resource sharing to true, that behavior wont change. But making it optional makes sense, default we want is indeed inherit whatever parent has.